app/assets/javascripts/discourse/app/components/sidebar/toggle-all-sections.gjs

```diff
@@ -0,0 +1,54 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import { ADMIN_NAV_MAP } from "discourse/lib/sidebar/admin-nav-map";
+
+export default class ToggleAllSections extends Component {
+  @service sidebarState;
+  @service keyValueStore;
+  @tracked collapsedSections = this.sidebarState.collapsedSections;
+
+  get allSectionsExpanded() {
+    return ADMIN_NAV_MAP.every((adminNav) => {
```

Comment:
I wonder if we can pass down a reference to `sections` in the sidebar from admin-header.gjs (and from the parent component to that?). If not thats fine

Reply:
it is tricky because we are passing classes around and later they are initialized

https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/app/components/sidebar/api-section.js#L10

If I pass this config down to toggle-all-sections I would need to create objects. I only need a name so maybe using ADMIN_NAV_MAP is good enough for now

Reply:
Ahhh sorry I forgot about this class thing in sidebar, all good :+1: